Documentation/ApiOverview/Events/Events/SEO/ModifyUrlForCanonicalTagEvent.rst

```diff
@@ -14,7 +14,26 @@ emptied.
 Example
 =======
 
-..  include:: /_includes/EventsContributeNote.rst.txt
+Registration of the event in the :file:`Services.yaml`:
+
+.. code-block:: yaml
+
+  MyVendor\MyPackage\EventListener\MyEventListener:
+    tags:
+      - name: event.listener
+        identifier: 'my-package/my-listener'
+        event: TYPO3\CMS\Seo\Event\ModifyUrlForCanonicalTagEvent
+
+.. code-block:: php
+
+    final class MyEventListener {
+        public function __invoke(ModifyUrlForCanonicalTagEvent $event): void
+        {
+            // do your magic
+            // Note: $event->getUrl(); is initialized with the empty string value ''
```

Comment:
Don't really understand the comment: So `getUrl()` always returns an empty string? Or what is meant with "is initialized"?

Reply:
[https://github.com/TYPO3/typo3/blob/main/typo3/sysext/seo/Classes/Canonical/CanonicalGenerator.php#L52](https://github.com/TYPO3/typo3/blob/main/typo3/sysext/seo/Classes/Canonical/CanonicalGenerator.php#L52)
The Event is dispatched with an empty string and it is natively not called somewhere else. 
I first thought that $event->getUrl() contains the generated canonical url which can be modified. But thats not the case.

Reply:
Ah okay, so it is more of a "before" event. The Core logic kicks in only when the event returns also an empty URL.